substrate/frame/vesting/src/lib.rs

```diff
@@ -678,6 +686,119 @@ impl<T: Config> Pallet<T> {
+///   (`schedules_left` - `starting_block`) and the per block amount when the locked amount is
+///   divisible by the per block amount.
+/// * However, If the locked amount is not divisible by the per block amount, the final vesting
+///   block (`schedules_left` - 1), the unvested amount should be equal to the remainder.
+///
+/// `handle_during_schedule`:
+/// * the amount `still_vesting` must be equal to the product of the remaining blocks to vest
+///   (`schedules_left` - `current_block`) and per block amount when the locked amount is divisible
+///   by the per block amount.
+/// * However, If the locked amount is not divisible by the per block amount, then at the final
+///   vesting block of the current schedule (`schedules_left` - 1), the unvested amount should be
+///   equal to the remainder.
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	pub fn do_try_state() -> Result<(), TryRuntimeError> {
+		for (_, d) in Vesting::<T>::iter() {
+			let infos = d.to_vec();
+
+			for info in infos.iter() {
+				let schedules_left: BalanceOf<T> =
+					info.ending_block_as_balance::<T::BlockNumberToBalance>();
+				let starting_block = T::BlockNumberToBalance::convert(info.starting_block());
+				let current_block_to_balance = T::BlockNumberToBalance::convert(
+					T::BlockNumberProvider::current_block_number(),
+				);
+
+				if current_block_to_balance < starting_block {
+					Self::handle_before_schedule_starts(info, starting_block, schedules_left)?;
+				} else {
+					Self::handle_during_schedule(info, current_block_to_balance, schedules_left)?;
+				}
+			}
+		}
+		Ok(())
+	}
+
+	fn handle_before_schedule_starts(
+		info: &VestingInfo<BalanceOf<T>, BlockNumberFor<T>>,
+		starting_block: BalanceOf<T>,
+		schedules_left: BalanceOf<T>,
+	) -> Result<(), TryRuntimeError> {
+		let count = schedules_left.saturating_sub(starting_block);
+
+		if (info.locked() % info.per_block()).is_zero() {
+			ensure!(
+                info.locked_at::<T::BlockNumberToBalance>(T::BlockNumberProvider::current_block_number()) == (count * info.per_block()),
+                TryRuntimeError::Other("Before schedule starts, the vesting balance should be equal to the total per block releases")
+            );
+		} else {
+			let re = info.locked() % info.per_block();
+
+			let final_vest_block =
+				schedules_left.saturating_sub(One::one()).saturated_into::<u64>();
+
+			let final_vest_amount =
+				info.locked_at::<T::BlockNumberToBalance>(final_vest_block.saturated_into());
+
+			ensure!(final_vest_amount == re, TryRuntimeError::Other("Before schedule starts, the final vest amount should be equal to the remainder"));
+
+			let no_schedules_left = schedules_left.saturated_into::<u64>();
+
+			let no_locks =
+				info.locked_at::<T::BlockNumberToBalance>(no_schedules_left.saturated_into());
+
+			ensure!(
+				no_locks == Zero::zero(),
+				TryRuntimeError::Other("After all schedules, all amounts should be unlocked")
+			);
```

Comment:
Can you please also check that all accounts have at least the required amounts locked in their Account?  
Should be queryble by `Currency::locked` or something.

Reply:
the locked amount does not seem to be directly queryable, using `vesting_balance` should suffice?

Reply:
`Currency::locks`, this seems only queryable in test?

Reply:
`vesting_balance` provides a quick query on user locked balances